pennylane/qchem/structure.py

```diff
@@ -412,3 +413,136 @@ def excitations_to_wires(singles, doubles, wires=None):
+                [-0.86795121,  0.7320799 , -1.7236192 ]]),
+        0)
+
+        >>> mol_data_from_pubchem("VNWKTOKETHGBQD-UHFFFAOYSA-N", "InChIKey")
+        (['C', 'H', 'H', 'H', 'H'],
+        array([[ 0.        ,  0.        ,  0.        ],
+                [ 1.04709725,  1.51102501,  0.93824902],
+                [ 1.29124986, -1.53710323, -0.47923455],
+                [-1.47058487, -0.70581271,  1.26460472],
+                [-0.86795121,  0.7320799 , -1.7236192 ]]),
+        0)
+
+    """
+
+    try:
+        # pylint: disable=import-outside-toplevel, unused-import, multiple-imports
+        import pubchempy as pcp
+    except ImportError as Error:
+        raise ImportError(
+            "This feature requires pubchempy.\nIt can be installed with: pip install pubchempy."
+        ) from Error
+
+    # https://gist.github.com/lsauer/1312860/264ae813c2bd2c27a769d261c8c6b38da34e22fb#file-smiles_inchi_annotated-js
+    identifier_patterns = {
+        "name": re.compile(r"^[a-zA-Z0-9_+-]+$"),
+        "cas": re.compile(r"^\d{1,7}\-\d{2}\-\d$"),
+        "smiles": re.compile(
+            r"^(?!InChI=)(?!\d{1,7}\-\d{2}\-\d)(?![A-Z]{14}\-[A-Z]{10}(\-[A-Z])?)[^J][a-zA-Z0-9@+\-\[\]\(\)\\\/%=#$]{1,}$"
+        ),
+        "inchi": re.compile(
+            r"^InChI\=1S?\/[A-Za-z0-9\.]+(\+[0-9]+)?(\/[cnpqbtmsih][A-Za-z0-9\-\+\(\)\,\/\?\;\.]+)*$"
+        ),
+        "inchikey": re.compile(r"^[A-Z]{14}\-[A-Z]{10}(\-[A-Z])?"),
+    }
+    if identifier_type.lower() == "cid":
+        cid = int(identifier)
+    else:
+        if identifier_type.lower() not in identifier_patterns.keys():
+            raise ValueError(
+                "Specified identifier type is not supported. Supported type are: name, CAS, SMILES, InChI, InChIKey."
+            )
+        try:
+            if identifier_patterns[identifier_type.lower()].search(identifier):
+                if identifier_type.lower() == "cas":
+                    identifier_type = "name"
+                cid = pcp.get_cids(identifier, namespace=identifier_type.lower())[0]
+            else:
+                raise ValueError(
+                    f"Specified identifier doesn't seem to match type: {identifier_type}."
+                )
+        except (IndexError, pcp.NotFoundError) as exc:
+            raise ValueError("Specified molecule does not exist in the PubChem Database.") from exc
+
+    try:
+        pcp_molecule = pcp.Compound.from_cid(cid, record_type="3d")
+    except pcp.NotFoundError:
+        pcp_molecule = pcp.Compound.from_cid(cid, record_type="2d")
+    except ValueError as exc:
+        raise ValueError("Provided CID (or Identifier) is None.") from exc
+
+    mol_data = pcp_molecule.to_dict(properties=["atoms", "charge"])
+    symbols = [atom["element"] for atom in mol_data["atoms"]]
+    geometry = (
+        np.array([[atom["x"], atom["y"], atom.get("z", 0.0)] for atom in mol_data["atoms"]])
+        / bohr_angs
+    )
+    charge = mol_data["charge"]
```

Comment:
I think we can avoid returning the charge. This is something that should be obvious to the user. I know the charge of H3+ and OH-